Scenes/Tavern/Publician.gd

```diff
@@ -0,0 +1,199 @@
+extends Node2D
+
+onready var item_list = get_parent().find_node("ItemList")
+onready var dwarver_cion = load("res://Scenes/Tavern/dwarver_icon.png")
+var NegligibleInformScreen = load("res://Scenes/Screens/NegligibleInform/NegligibleInform.tscn")
+
+const SAVE_PATH = "res://quests.json"
+
+var quests: Array = []
+var selected_quest: int = 0
+
+
+
+func _ready():
+	add_to_group('IHaveSthToSave')
+	load_quests()
+
+func _process(delta):
+	save_quests()
+	if item_list.get_selected_items().size()>0:
+		selected_quest = item_list.get_selected_items()[0]
+	
+func load_quests():
+	var save_file = File.new()
+	if not save_file.file_exists(SAVE_PATH):
+		print("Nie znaleziono pliku!!!")
+		create_default_quests()
+		return
+	
+	save_file.open(SAVE_PATH, File.READ)
+	var data = JSON.parse(save_file.get_as_text()).result;
+
+	for index in data.keys():
+		if data[index]["type"] == QuestType.kill_dwarver:
+			quests.append(QuestKill.new(data[index]["count"]))
+		if data[index]["type"] == QuestType.spend_xp:
+			quests.append(QuestSpendXp.new(data[index]["count"]))
+		if data[index]["type"] == QuestType.spend_gold:
+			quests.append(QuestSpendXp.new(data[index]["count"]))
+	
+	load_list_panel()
+
+func save_quests():
+	var save_dict = { }
+	for quest in quests:
+		save_dict[quests.find(quest)] = quest.data
+	
+	var save_file = File.new()
+	save_file.open(SAVE_PATH, File.WRITE)
+	save_file.store_line(to_json(save_dict))
+	save_file.close()
+
+func create_default_quests():
+	add_kill_dwarves_quest()
+	add_spend_gold_quest()
+	add_kill_dwarves_quest()
+	add_spend_xp_quest()
+	add_spend_xp_quest()
+	add_spend_gold_quest()
+	add_kill_dwarves_quest()
+	add_spend_gold_quest()
+	add_kill_dwarves_quest()
+	add_spend_xp_quest()
+	add_spend_xp_quest()
+	add_spend_gold_quest()
+
+func load_list_panel():
+	for index in quests.size():
+		item_list.add_item(String(quests[index].data["title"]),dwarver_cion)
+
+class QuestKill:
+	var data = {
+		title = "Zabij x krasnali",
+		gold_reward = 100.0,
+		xp_reward = 40.0,
+		count = 4,
+		amount = 0,
+		type = QuestType.kill_dwarver
+		}
+	
+	func _init(Count):
+		data["count"] = Count
+		data["title"] = "Zabij krasnale: "+String(Count)	
+		data["gold_reward"] =  stepify(float(Count) * float(Count) * randf()*2,2)
+		data["xp_reward"] = stepify(Count * (randf()*6+1),2)
+
+class QuestSpendXp:
```

Comment:
Why not to use inheritance? We creating three, almost the same classes with other init function and other `data["type"]` value (that we can set in init function). 
Here inheritance examples in gdscript: 
1. https://www.reddit.com/r/godot/comments/54w6at/extending_classes_and_variables/
1. https://docs.godotengine.org/en/3.1/getting_started/scripting/gdscript/gdscript_basics.html#inheritance